src/libraries/System.Private.CoreLib/src/System/Span.cs

```diff
@@ -300,6 +300,15 @@ public unsafe void Clear()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public unsafe void Fill(T value)
         {
+            // Mono works faster with Unsafe.InitBlockUnaligned
+#if MONO
+            if (sizeof(T) == 1)
+            {
+                Unsafe.InitBlockUnaligned(ref Unsafe.As<T, byte>(ref _reference), *(byte*)&value, (uint)_length);
+                return;
+            }
+#endif
```

Comment:
Is it worth doing this over updating Mono to recognize/special-case the same scenarios as dotnet/runtime?

At the very least, should we ensure an issue tracking that same support is added so this `#if MONO` can be removed long term?

Reply:
Not sure whether Mono folks want to add more intrinsics in Mini and Interpreter considering this short path just works as is, so leaving that decision up to them, cc @BrzVlad

Reply:
I'm not a big fan of having MONO ifdefs in SPC. Also I think the regression is limited to interpreter, so disabling for all mono might be uncalled for. Also this should be easy to intrinsify, I'll take a look.

Reply:
> I'm not a big fan of having MONO ifdefs in SPC. Also I think the regression is limited to interpreter, so disabling for all mono might be uncalled for. Also this should be easy to intrinsify, I'll take a look.

The regression can be seen for mini (JIT) as well. 
![image](https://github.com/dotnet/runtime/assets/55735845/061c0c02-858c-42c5-95fd-0b4c9e2b1ab1)

Most likely for AOT as well since, the microbenchmark is written using generics, however, our AOT perf lines are currently broken. 

Having Mono ifdefs in managed code doesn't seem to be a good long-term solution. If we go with this solution and the solution proves to remove the regression, I think we should add these intrinsics to Mono. What do you think about this @fanyang-mono?

Reply:
I took a look a quick look at the original PR (https://github.com/dotnet/runtime/pull/98623). It seems to me that to intrinsify `SpanHelpers.Memmove`, the code path for intrinsifying `Buffer.Memmove` was used with some tweak. Mono intrinsify `Buffer.Memmove` as well. See the code below.
https://github.com/dotnet/runtime/blob/fd48b6f5d1ff81a81d09e9d72982cc9e8d139852/src/mono/mono/mini/intrinsics.c#L288-L310

I think the proper fix should be updating the above code as CoreCLR did.

Reply:
Created a github issue for it #99161